src/App.tsx

```diff
@@ -15,49 +17,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+const getReorderedGoods = (
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) => {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((g1, g2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return g1.localeCompare(g2);
+      case SortType.LENGTH:
+        return g1.length - g2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setReversed] = useState(false);
+
+  const reverse = () => {
+    setReversed(!isReversed);
+  };
+
+  const sortByName = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const reset = () => {
+    setReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const goodsToRender
+      = getReorderedGoods(goodsFromServer, sortType, isReversed);
+
+  const wasItSorted = sortType === SortType.NONE && isReversed === false;
```

Comment:
```suggestion
  const isSorted = sortType !== SortType.NONE && isReversed;
```
it is better to return `true` from the expression if you have a such boolean check, it can be confusing

Reply:
const isSorted = sortType !== SortType.NONE && isReversed;
this suggestion isnt work in this way, it shoul be opposite, by the way I done it correct)